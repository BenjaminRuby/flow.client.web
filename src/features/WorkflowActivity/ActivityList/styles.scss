@import "~@boomerang/boomerang-styles/lib/colors";

.c-activity-list {
  display: flex;
  flex-direction: column;
  margin: 1.5rem 0rem;
}
.bmrg--c-loading-animation.s-activities-loading {
  height: 3rem;
  justify-content: center;
}
.s-activities-loading {
  .bmrg--b-loading-animation__text {
    display: none;
  }
  .bmrg--b-loading-animation__img {
    transform: scale(0.3);
  }
}

.b-activities__scroll-up-button {
  background-color: red;
  width: 2rem;
  height: 2rem;
  position: fixed;
}
.c-activities__scroll-up {
  opacity: 0.01;
  position: fixed;
  bottom: 2rem;
  right: 5rem;
  transition: ease-in-out 150ms;

  &.--scroll-up-togled {
    transition: ease-in-out 150ms;
    cursor: pointer;
    opacity: 1;
  }
}
.b-activities__scroll-button {
  width: 2rem;
  height: 2rem;
  margin-bottom: 0.75rem;
  transform: rotate(270deg);
}
.b-activities__scroll-label {
  color: $cerulean-70;
  font-size: 1rem;
  font-weight: 600;
}
