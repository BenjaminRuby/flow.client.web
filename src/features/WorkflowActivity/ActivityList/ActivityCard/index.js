import React, { Component } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import moment from "moment";
import getHumanizedDuration from "@boomerang/boomerang-utilities/lib/getHumanizedDuration";
import { ACTIVITY_STATUSES_TO_TEXT, ACTIVITY_STATUSES_TO_ICON } from "Constants/activityStatuses";
import imgs from "Assets/icons";
import "./styles.scss";

class ActivityCard extends Component {
  static propTypes = {
    activity: PropTypes.object.isRequired,
    match: PropTypes.object.isRequired
  };

  render() {
    const {
      shortDescription,
      creationDate,
      icon,
      duration = 0,
      id,
      initiatedByUserName,
      status,
      teamName,
      trigger,
      workflowId,
      workflowName
    } = this.props.activity;
    return (
      <Link
        className={`c-activities-card --${status}`}
        to={{
          pathname: `/activity/${workflowId}/execution/${id}`,
          state: { fromUrl: this.props.match.url, fromText: "Activity" }
        }}
      >
        <div className="c-activity-card-workflow">
          <div className="c-activity-card-section--left">
            <div className="c-activity-card__icon">
              <img className="b-activity-card__icon" src={imgs[icon ? icon : "docs"]} alt="Worklflow icon" />
            </div>
            <div className="c-activity-card-workflow-info">
              <label className="b-activity-card__name">{workflowName}</label>
              <label className="b-activity-card__description">{shortDescription}</label>
            </div>
          </div>
          <div className="c-activity-card-section--right">
            <div className="c-activity-card-execution-info --first">
              <section className="b-activity-card__row">
                <p className="b-activity-card__label">Team</p>
                <p className="b-activity-card__data">{teamName}</p>
              </section>
              <section className="b-activity-card__row">
                <p className="b-activity-card__label">Trigger</p>
                <p className="b-activity-card__data">
                  {trigger ? trigger.slice(0, 1).toUpperCase() + trigger.slice(1) : "---"}
                </p>
              </section>
              <section className="b-activity-card__row">
                <p className="b-activity-card__label">Initiated by</p>
                <p className="b-activity-card__data">{initiatedByUserName || "---"}</p>
              </section>
            </div>
            <div className="c-activity-card-execution-info">
              <section className="b-activity-card__row">
                <p className="b-activity-card__label">Start Time</p>
                <p className="b-activity-card__data">{moment(creationDate).format("YYYY-MM-DD hh:mm A")}</p>
              </section>
              <section className="b-activity-card__row">
                <p className="b-activity-card__label">Duration</p>
                <p className="b-activity-card__data">
                  {duration ? getHumanizedDuration(parseInt(duration / 1000, 10)) : "---"}
                </p>
              </section>
              <section className="b-activity-card__row">
                <p className="b-activity-card__label"> Status</p>
                <p className="b-activity-card__data">
                  <img
                    className="b-activity-card__status-icon"
                    src={ACTIVITY_STATUSES_TO_ICON[status ? status : "notstarted"]}
                    alt={`Status ${status}`}
                  />
                  {ACTIVITY_STATUSES_TO_TEXT[status ? status : "notstarted"]}
                </p>
              </section>
            </div>
          </div>
        </div>
      </Link>
    );
  }
}

export default ActivityCard;
