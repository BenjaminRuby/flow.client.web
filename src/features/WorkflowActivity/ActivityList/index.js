import React, { Component } from "react";
import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";
import { LoadingAnimation } from "@boomerang/carbon-addons-boomerang-react";
import NoDisplay from "@boomerang/boomerang-components/lib/NoDisplay";
import InfiniteScroll from "react-infinite-scroller";
import ActivityCard from "./ActivityCard";
import ScrollUp from "Components/ScrollUp";
import "./styles.scss";

class ActivityList extends Component {
  static propTypes = {
    activities: PropTypes.array.isRequired,
    history: PropTypes.object.isRequired,
    match: PropTypes.object.isRequired,
    loadMoreActivities: PropTypes.func.isRequired,
    isLoading: PropTypes.bool,
    hasMoreActivities: PropTypes.bool
  };

  loadMoreActivities = () => {
    this.props.loadMoreActivities();
  };

  render() {
    const { activities, hasMoreActivities, history, isLoading, match, location } = this.props;

    return (
      <>
        <InfiniteScroll
          className="c-activity-list"
          pageStart={0}
          loadMore={this.loadMoreActivities}
          hasMore={hasMoreActivities && !isLoading}
          loader={<LoadingAnimation className="s-activities-loading" />}
          useWindow={true}
        >
          {!activities.length && !hasMoreActivities && !isLoading ? (
            <NoDisplay
              style={{ marginTop: "4rem" }}
              text="Looks like you need to run some workflows!"
              textLocation="below"
            />
          ) : (
            activities.map(activity => {
              return (
                <ActivityCard
                  activity={activity}
                  history={history}
                  key={activity.id}
                  match={match}
                  location={location}
                />
              );
            })
          )}
        </InfiniteScroll>
        <ScrollUp />
      </>
    );
  }
}

export default withRouter(ActivityList);
