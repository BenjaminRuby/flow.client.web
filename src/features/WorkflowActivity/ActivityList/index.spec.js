import React from "react";
import ActivityList from "./index";

const history = {};
const activities = [
  {
    creationDate: 1541070397344,
    duration: 13886,
    id: "5bdade3d37dc4700011b06c3",
    initiatedBy: "5bc7b126f7856000012cd95d",
    status: "completed",
    workflowId: "5bd9d8ab7eb44d0001772e64",
    workflowRevisionid: "5bd9e1cfa40f5d0001249bfa",
    description: "file ingestion",
    teamName: "Isa's Team",
    icon: "flow",
    shortDescription: "file ingestion",
    workflowName: "Business Process Test"
  }
];

const match = {
  url: "test.url"
};

const mockfn = jest.fn();

const props = {
  activities,
  history,
  match,
  loadMoreActivities: mockfn,
  setMoreActivities: mockfn
};

describe("ActivityList --- Snapshot", () => {
  it("Capturing Snapshot of ActivityList", () => {
    const { baseElement } = rtlRouterRender(<ActivityList {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
