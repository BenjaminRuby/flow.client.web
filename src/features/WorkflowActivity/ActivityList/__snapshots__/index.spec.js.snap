// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityList --- Snapshot Capturing Snapshot of ActivityList 1`] = `
<body>
  <div>
    <div
      class="c-activity-list"
    >
      <a
        class="c-activities-card --completed"
        href="/activity/5bd9d8ab7eb44d0001772e64/execution/5bdade3d37dc4700011b06c3"
      >
        <div
          class="c-activity-card-workflow"
        >
          <div
            class="c-activity-card-section--left"
          >
            <div
              class="c-activity-card__icon"
            >
              <img
                alt="Worklflow icon"
                class="b-activity-card__icon"
                src="flow.svg"
              />
            </div>
            <div
              class="c-activity-card-workflow-info"
            >
              <label
                class="b-activity-card__name"
              >
                Business Process Test
              </label>
              <label
                class="b-activity-card__description"
              >
                file ingestion
              </label>
            </div>
          </div>
          <div
            class="c-activity-card-section--right"
          >
            <div
              class="c-activity-card-execution-info --first"
            >
              <section
                class="b-activity-card__row"
              >
                <p
                  class="b-activity-card__label"
                >
                  Team
                </p>
                <p
                  class="b-activity-card__data"
                >
                  Isa's Team
                </p>
              </section>
              <section
                class="b-activity-card__row"
              >
                <p
                  class="b-activity-card__label"
                >
                  Trigger
                </p>
                <p
                  class="b-activity-card__data"
                >
                  ---
                </p>
              </section>
              <section
                class="b-activity-card__row"
              >
                <p
                  class="b-activity-card__label"
                >
                  Initiated by
                </p>
                <p
                  class="b-activity-card__data"
                >
                  ---
                </p>
              </section>
            </div>
            <div
              class="c-activity-card-execution-info"
            >
              <section
                class="b-activity-card__row"
              >
                <p
                  class="b-activity-card__label"
                >
                  Start Time
                </p>
                <p
                  class="b-activity-card__data"
                >
                  2019-01-01 12:00 AM
                </p>
              </section>
              <section
                class="b-activity-card__row"
              >
                <p
                  class="b-activity-card__label"
                >
                  Duration
                </p>
                <p
                  class="b-activity-card__data"
                >
                  13 seconds
                </p>
              </section>
              <section
                class="b-activity-card__row"
              >
                <p
                  class="b-activity-card__label"
                >
                   Status
                </p>
                <p
                  class="b-activity-card__data"
                >
                  <img
                    alt="Status completed"
                    class="b-activity-card__status-icon"
                    src="passed.svg"
                  />
                  Succeeded
                </p>
              </section>
            </div>
          </div>
        </div>
      </a>
    </div>
    <aside
      aria-label="Scroll to top of page"
      class="c-activities__scroll-up "
      data-testid="react-scroll-up-button"
      role="button"
      tabindex="-1"
    >
      <div
        style="display: flex; flex-direction: column; align-items: center;"
      >
        <img
          alt="chevron"
          class="b-activities__scroll-button"
          src="doublechevron.svg"
        />
        <div
          class="b-activities__scroll-label"
        >
          Go to top
        </div>
      </div>
    </aside>
  </div>
</body>
`;
